lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -405,6 +379,41 @@ def log_pruning(kubectl_output)
       @logger.summary.add_action("pruned #{pruned.length} #{'resource'.pluralize(pruned.length)}")
     end
 
+    def record_apply_failure(err)
+      warn_msg = "WARNING: Any resources not mentioned in the error(s) below were likely created/updated. " \
+        "You may wish to roll back this deploy."
+      @logger.summary.add_paragraph(ColorizedString.new(warn_msg).yellow)
+
+      unidentified_errors = []
```

Comment:
This might be a bad idea. If there were a multi-line error message where one of the middle lines has a filename in it, we'd be splitting it up strangely. I haven't seen that, but I certainly haven't seen everything. An alternative is to always dump the complete error before the individual cases we were able to extract and enhance.